django/contrib/gis/gdal/raster/source.py

```diff
@@ -47,19 +61,68 @@ def __init__(self, ds_input, write=False):
         self._write = 1 if write else 0
         Driver.ensure_registered()
 
+        # Preprocess json inputs. This converts json strings to dictionaries,
+        # which are parsed below the same way as direct dictionary inputs.
+        if isinstance(ds_input, six.string_types) and json_regex.match(ds_input):
+            ds_input = json.loads(ds_input)
+
         # If input is a valid file path, try setting file as source.
         if isinstance(ds_input, six.string_types):
-            if os.path.exists(ds_input):
-                try:
-                    # GDALOpen will auto-detect the data source type.
-                    self.ptr = capi.open_ds(force_bytes(ds_input), self._write)
-                except GDALException as err:
-                    raise GDALException('Could not open the datasource at "{}" ({}).'.format(
-                        ds_input, err))
-            else:
-                raise GDALException('Unable to read raster source input "{}"'.format(ds_input))
+            if not os.path.exists(ds_input):
+                raise GDALException(
+                    'Unable to read raster source input "{}"'.format(ds_input))
+            try:
+                # GDALOpen will auto-detect the data source type.
+                self._ptr = capi.open_ds(force_bytes(ds_input), self._write)
+            except GDALException as err:
+                raise GDALException(
+                    'Could not open the datasource at "{}" ({}).'.format(ds_input, err))
```

Comment:
Actually I would probably keep most of those exceptions as a single line -- we prefer long erlines when it improves readability.